app/packages/looker/src/worker/canvas-decoder.ts

```diff
@@ -70,8 +72,9 @@ export const recastBufferToMonoChannel = (
 export const decodeWithCanvas = async (blob: Blob, cls: string) => {
   let channels: number = 4;
 
-  if (blob.type === "image/png") {
-    const colorType = await getPngcolorType(blob);
+  if (blob.type !== "image/jpg" && blob.type !== "image/jpeg") {
```

Comment:
This implies we still rely on MIME type, specifically for jpeg?

Reply:
Not really... more like a heuristic. We don't want to unnecessarily check if it's PNG if the mime type says it's a jpeg.

Note that all of this is used just to determine number of channels.